### DISCUSSION TITLE
Env prefix support when explicit env name is passed to Field

### DISCUSSION THREAD

**Discussion by Faylixe** [pydantic/pydantic#2642] [Feature Request]
Maybe this is a voluntary aspect of environment vars based Settings. But actually if an `env` parameter is given to a `Field` then the provided configuration prefix is not taken into account. I feel like there is either a lack of documentation about it or this problem was expected in the first place.

To be more specific here is what I went through:

- Design a custom `BaseSettings` class for my application.
- Named all field in lowercase.
- Added a prefix to scope my application envvars.
- Considered that envvars will be uppercase (which is the natural convention usually).
- Discovered that case sensitivity is an issue.
- Instead of renaming my field name, just switch to `Field` declaration with `env` parameter (without prefix) in uppercase.
- Finally read Pydantic source code to see that prefix is not considered when using `env` parameter.

I do not feel like it should behave this way, as said natural convention for environment variable is to use uppercase naming scheme, but I do not necessary want my setting class to be using uppercase field name (this can be obviously counter argued by saying that after all, those values are constant setting values and thus can be legitimately uppercased, which I would respect :D).

I would gladly work on a PR for either updating documentation to add a small warning section about it, or fixing it.

**Comment by nymous** [comment]
Hello!

I don't know about this issue with `env` and the prefix, but BaseSettings is case-**in**sensitive by default.

```py
from pydantic import BaseSettings

class Settings(BaseSettings):
    user: str

    class Config:
        env_prefix = "my_prefix_"

settings = Settings()
print(settings)
```

```bash
$ my_prefix_USER=nymous python main.py
user='nymous'

$ MY_PREFIX_USER=nymous python main.py
user='nymous'

$ my_prefix_user=nymous python main.py
user='nymous'
```

**Reply by Faylixe** [reply]
Yes indeed I actually misconfigured the prefix (missing the additional `_` at the end and wrongly accuse the feature 🙈) although the fact that the `env` parameter override any defined prefix in the configuration is still misleading and a warning note in the documentation won't do any harm, I will PR the docs then :)
